azext_edge/edge/providers/check/mq.py

```diff
@@ -539,14 +625,349 @@ def evaluate_brokers(
+                detail_level=detail_level,
+            )
+
+    return check_manager.as_dict(as_list)
+
+
+def evaluate_broker_authorizations(
+    as_list: bool = False,
+    detail_level: int = ResourceOutputDetailLevel.summary.value,
+    resource_name: str = None,
+) -> Dict[str, Any]:
+
+    check_manager = CheckManager(
+        check_name="evalBrokerAuthorizations",
+        check_desc="Evaluate MQTT Broker Authorizations",
+    )
+
+    target_authorizations = "brokerauthorizations.mqttbroker.iotoperations.azure.com"
+    authz_conditions = ["spec.authorizationPolicies"]
+
+    all_authorizations = get_resources_by_name(
+        api_info=MQ_ACTIVE_API,
+        kind=MqResourceKinds.BROKER_AUTHORIZATION,
+        resource_name=resource_name,
+    )
+
+    if not all_authorizations:
+        fetch_authorizations_error_text = (
+            f"Unable to fetch {MqResourceKinds.BROKER_AUTHORIZATION.value}s in any namespace."
+        )
+        check_manager.add_target(target_name=target_authorizations)
+        check_manager.add_target_eval(
+            target_name=target_authorizations,
+            status=CheckTaskStatus.skipped.value,
+            value=fetch_authorizations_error_text,
+        )
+        check_manager.add_display(
+            target_name=target_authorizations,
+            display=Padding(fetch_authorizations_error_text, (0, 0, 0, DEFAULT_PADDING)),
+        )
+        return check_manager.as_dict(as_list)
+
+    for namespace, authorizations in get_resources_grouped_by_namespace(all_authorizations):
+        check_manager.add_target(
+            target_name=target_authorizations,
+            namespace=namespace,
+        )
+        check_manager.add_display(
+            target_name=target_authorizations,
+            namespace=namespace,
+            display=Padding(
+                f"Broker Authorizations in namespace {{[purple]{namespace}[/purple]}}", (0, 0, 0, DEFAULT_PADDING)
+            ),
+        )
+
+        authorizations = list(authorizations)
+        added_broker_ref_condition = False
+
+        for authz in authorizations:
+            authz_metadata = authz["metadata"]
+            authz_name = authz_metadata["name"]
+
+            # check broker reference
+            broker_ref = authz_metadata.get("ownerReferences", [])
+            ref_display = _evaluate_broker_reference(
+                check_manager=check_manager,
+                owner_reference=broker_ref,
+                target_name=target_authorizations,
+                namespace=namespace,
+                resource_name=authz_name,
+                added_condition=added_broker_ref_condition,
+            )
+
+            authz_desc = f"\n- Broker Authorization {{{colorize_string(authz_name)}}}. {ref_display}"
+            check_manager.add_display(
+                target_name=target_authorizations,
+                namespace=namespace,
+                display=Padding(authz_desc, (0, 0, 0, DEFAULT_PADDING)),
+            )
```

Comment:
Similar question here to the previous one about `evaluate_broker_ref`: https://github.com/Azure/azure-iot-ops-cli-extension/pull/382/files#r1868441670